### PR TITLE
Memif bench test

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: Denis Tingajkin denis.tingajkin@xored.com

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [x] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -320,6 +320,12 @@ workflows:
           test_tags: "recover"
           requires:
             - packet-deploy-k8s-2
+      - packet-integration-tests:
+          name: "test benchmark"
+          cluster_id: "1"
+          test_tags: "bench"
+          requires:
+            - packet-deploy-k8s-1
```
Comment (nickolaev) [inline .circleci/config.yml]: Like this the CI will try to run in parrallel "test basic usecase" and "test basic usecase".
Better to put here 
```yaml
      - packet-integration-tests:
	          name: "test benchmark"
	          cluster_id: "2"
	          test_tags: "bench"
	          requires:
	            - packet-deploy-k8s-1
	            - "test recover"
```

Ther recover tests are shorter, so queuing the benchmark on the same cluster just after them makes sense.

Comment (denis-tingajkin) [reply .circleci/config.yml]: thanks, fixed.

Comment (nickolaev) [reply .circleci/config.yml]: Sorry, I should have been more accurate, you actually need:
```yaml
      - packet-integration-tests:
	          name: "test benchmark"
	          cluster_id: "2"
	          test_tags: "bench"
	          requires:
	            - "test recover"
```

File: test/integration/bench_memif_test.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
+package nsmd_integration_tests
```
Comment (nickolaev) [inline test/integration/bench_memif_test.go]: You need to have this line in the beginning of the file to make it run during the basic tests phase:
```golang
// +build basic
```

File: test/integration/bench_memif_test.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+	k8s, err := kube_testing.NewK8s()
+	defer k8s.Cleanup()
+
+	Expect(err).To(BeNil())
+
+	k8s.PrepareDefault()
+
+	node := createNode(k8s)
+	vppAgentIcmp := k8s.CreatePod(pods.VppagentICMPResponderPod(icmpAgentName, node, icmpEnv()))
```
Comment (nickolaev) [inline test/integration/bench_memif_test.go]: Please check test_utils for a universal pod creation.

File: test/integration/bench_memif_test.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+	k8s, err := kube_testing.NewK8s()
+	defer k8s.Cleanup()
+
+	Expect(err).To(BeNil())
+
+	k8s.PrepareDefault()
+
+	node := createNode(k8s)
+
+	vppAgentIcmp := k8s.CreatePod(pods.VppagentICMPResponderPod(icmpAgentName, node, nscEnv()))
```
Comment (nickolaev) [inline test/integration/bench_memif_test.go]: Please check test_utils for a universal pod creation.

File: test/integration/bench_memif_test.go (hunk 2 above)
Comment (denis-tingajkin) [reply test/integration/bench_memif_test.go]: I just testing memif_test on CI, I've marked this PR as WIP. I'll review all comments later

File: .circleci/config.yml (hunk 1 above)
Comment (denis-tingajkin) [reply .circleci/config.yml]: thanks, fixed

File: test/integration/bench_memif_test.go (hunk 2 above)
Comment (denis-tingajkin) [reply test/integration/bench_memif_test.go]: Fixed

File: test/integration/bench_memif_test.go (hunk 3 above)
Comment (denis-tingajkin) [reply test/integration/bench_memif_test.go]: fixed

File: test/integration/bench_memif_test.go [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+	nsc     *v1.Pod
+}
+
+func icmpEnv() map[string]string {
+	return map[string]string{
+		"ADVERTISE_NSE_NAME":   "icmp-responder",
+		"ADVERTISE_NSE_LABELS": "app=icmp",
+		"IP_ADDRESS":           "10.20.1.0/24",
+	}
+}
```
[the remark this answers is not in the document]
Comment (denis-tingajkin) [reply test/integration/bench_memif_test.go]: fixed

File: test/integration/bench_memif_test.go (hunk 4 above)
Comment (denis-tingajkin) [reply test/integration/bench_memif_test.go]: fixed

**Pull request merged**